src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,297 @@
+/**
+ * @file methods/kernel_svm/kernel_svm_impl.hpp
+ * @author Himanshu Pathak
+ *
+ * Implementation of Kernel SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+#define MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "kernel_svm.hpp"
+
+namespace mlpack {
+namespace svm {
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const double regularization,
+    const bool fitIntercept,
+    const size_t maxIter,
+    const double tol,
+    const KernelType kernel) :
+    regularization(regularization),
+    fitIntercept(fitIntercept),
+    kernel(kernel)
+{
+  intercept = 0;
+  Train(data, labels, maxIter, tol);
+}
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const double regularization,
+    const bool fitIntercept,
+    const KernelType kernel) :
+    regularization(regularization),
+    fitIntercept(fitIntercept),
+    kernel(kernel)
+{
+  intercept = 0;
+}
+
+template<typename MatType, typename KernelType>
+double KernelSVM<MatType, KernelType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t maxIter,
+    const double tol)
+{
+  trainCoefficients = arma::zeros(1, data.n_cols);
+
+  // Changing labels values to 1, -1 values provided
+  // by user should 0 and 1.
```

Comment:
Not sure why we have to transform the data here, but you can do this without a for loop: `trainCoefficients = labels * 2 - 1;`

Reply:
Previously I tried it but I was getting a memory error but should try this again.